### DISCUSSION TITLE
Way to pass local credentials for Remote Cache (from `turbo login`) to Docker build process?

### DISCUSSION THREAD

**Discussion by andyjy** [vercel/turborepo#1007] [Help] [answered]
E.g. can I somehow extract the `TURBO_TOKEN` obtained from `turbo login` and used for my local CLI, and pass it as an environment variable (thus though to my Docker build command, to be picked up in my Dockerfile wherein I wish to use `turbo run build`)?

### Use case
I have CI working successfully under GitHub actions as per the [continuous integration docs](https://turborepo.org/docs/guides/continuous-integration) - but when building docker images locally the build process inside my docker container doesn't have access to my local credentials for Turborepo Remote Caching.

### Workaround
Currently I've just manually configured a `TURBO_TOKEN` env var for local docker builds similarly to CI, but this duplicates `turbo login` and isn't easily portable to others.

### Things I tried
I tried inspecting the Turborepo source and found a reference to storing the token in my local config, but my local `<monorepo-root>/.turbo/config.json` only contains `teamId` and `ApiUrl`. The `turbo login` -> `turbo link` UX suggests to me the Remote Caching auth token will be stored somewhere globally...(?)

Thanks!

**Comment by sudoFerraz** [comment]
@andyjy On the docs it references a way to do any turbo command with the --api flag, this way you can change your DockerFile to not login, but use it from the env the token you've set

I also wish to continue using vercel remote cache on my CICD, but unfortunately on the docs it only references the linking step with a custom cache server https://turborepo.org/docs/features/remote-caching

Where can I find the official server url + auth token of the vercel build?

**Comment by anthonyshew** [comment]
The CI recipes show exactly this!

https://turbo.build/repo/docs/ci

**Reply by andyjy** [reply]
Thanks - unless I'm missing something in those CI docs, I think you may have misread my original question (or it was unclear). 

The CI docs show use of a TURBO_TOKEN env var to provide turbo credentials under CI (which makes sense given the fair assumption the CI credentials wouldn't change that often).

My original post asks how to achieve a similar outcome under a local development environment without having to manually set a local env var (which needs to be different for each individual). In particular I was hoping for a way to e.g. automatically set or extract the TURBO_TOKEN to the one obtained for the CLI by `turbo login`.

**Reply by anthonyshew** [reply]
Wow, I did read that completely wrong. 😆

@mehulkar gave you the right answer!

**Comment by mehulkar** [comment, marked as answer]
@andyjy this is an implementation detail, so I wouldn't rely on it (it is likely to change), but the token is written to disk. In 1.11, the location is based on the [config_dir()](https://docs.rs/dirs-next/latest/dirs_next/fn.config_dir.html) function [here](https://github.com/vercel/turbo/blob/943f856354fcd45af869b0dab56ec51144eb81cc/crates/turborepo-lib/src/commands/mod.rs#L83). In previous versions, you can dig through the Go code as well.
